napari/tests/test_viewer.py

```diff
@@ -54,254 +55,41 @@ def test_no_qt_loop():
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-    # Switch to 3D rendering mode and back to 2D rendering mode
-    viewer.dims.ndisplay = 3
-    assert viewer.dims.ndisplay == 3
-    viewer.dims.ndisplay = 2
-    assert viewer.dims.ndisplay == 2
-
-    # Run all class keybindings
-    for func in layer.class_keymap.values():
-        func(layer)
-
-    # Close the viewer
-    viewer.window.close()
-
-
-def test_add_volume(qtbot):
-    """Test adding volume."""
-    viewer = Viewer()
-    view = viewer.window.qt_viewer
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = np.random.random((10, 15, 20))
-    layer = viewer.add_image(data)
-    viewer.dims.ndisplay = 3
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 3
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-    # Switch to 3D rendering mode and back to 2D rendering mode
-    viewer.dims.ndisplay = 3
-    assert viewer.dims.ndisplay == 3
-    viewer.dims.ndisplay = 2
-    assert viewer.dims.ndisplay == 2
-
-    # Run all class keybindings
-    for func in layer.class_keymap.values():
-        func(layer)
-
-    # Close the viewer
-    viewer.window.close()
-
-
-def test_add_pyramid(qtbot):
-    """Test adding image pyramid."""
-    viewer = Viewer()
-    view = viewer.window.qt_viewer
-    qtbot.addWidget(view)
-
-    shapes = [(40, 20), (20, 10), (10, 5)]
     np.random.seed(0)
-    data = [np.random.random(s) for s in shapes]
-    layer = viewer.add_image(data, is_pyramid=True)
-    assert np.all(viewer.layers[0].data == data)
 
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-    # Switch to 3D rendering mode and back to 2D rendering mode
-    viewer.dims.ndisplay = 3
-    assert viewer.dims.ndisplay == 3
-    viewer.dims.ndisplay = 2
-    assert viewer.dims.ndisplay == 2
-
-    # Run all class keybindings
-    for func in layer.class_keymap.values():
-        func(layer)
-
-    # Close the viewer
-    viewer.window.close()
-
-
-def test_add_labels(qtbot):
-    """Test adding labels image."""
-    viewer = Viewer()
-    view = viewer.window.qt_viewer
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = np.random.randint(20, size=(10, 15))
-    layer = viewer.add_labels(data)
+    layer = viewer.add_layer_by_type(layer_type=layer_type, data=data)
+    if layer_type is layer_type.VOLUME:
```

Comment:
We shouldn't need the special casing for `Volume`. At the beginning of the test we should check `viewer.dims.ndisplay == 2` the default value, check all the other things are good, then switch to `viewer.dims.ndisplay=3`, check everything again, then switch back to `viewer.dims.ndisplay=2` and check everything again